tests/test_api/test_submit_job.py

```diff
@@ -299,6 +301,9 @@ def test_submit_bad_autorift_granule_names(client):
         # bad L8 sensor mode
         'LT08_L1GT_041001_20200125_20200925_02_T2',
         'LT09_L1GT_215109_20220125_20220125_02_T2',
+        # S2 name too long
+        'S2B_22XEQ_20190610_11_XL1C',
+        'S2A_22XEQ_20190610_10_XL1C',
```

Comment:
Might also include a test case of a name that's too short (e.g. exactly 22 characters)?